zcash_client_sqlite/src/wallet/common.rs

```diff
@@ -21,19 +21,47 @@ use crate::{
 };
 
 #[cfg(feature = "orchard")]
-use crate::ORCHARD_TABLES_PREFIX;
+use {crate::ORCHARD_TABLES_PREFIX, zcash_client_backend::data_api::ORCHARD_SHARD_HEIGHT};
+
+pub(crate) struct TableConstants {
+    pub(crate) table_prefix: &'static str,
+    pub(crate) output_index_col: &'static str,
+    pub(crate) output_count_col: &'static str,
+    pub(crate) note_reconstruction_cols: &'static str,
+    pub(crate) shard_height: u8,
+}
 
-pub(crate) fn per_protocol_names(
-    protocol: ShieldedProtocol,
-) -> (&'static str, &'static str, &'static str) {
-    match protocol {
-        ShieldedProtocol::Sapling => (SAPLING_TABLES_PREFIX, "output_index", "rcm"),
+const SAPLING_TABLE_CONSTANTS: TableConstants = TableConstants {
+    table_prefix: SAPLING_TABLES_PREFIX,
+    output_index_col: "output_index",
+    output_count_col: "sapling_output_count",
+    note_reconstruction_cols: "rcm",
+    shard_height: SAPLING_SHARD_HEIGHT,
+};
+
+#[cfg(feature = "orchard")]
+const ORCHARD_TABLE_CONSTANTS: TableConstants = TableConstants {
+    table_prefix: ORCHARD_TABLES_PREFIX,
+    output_index_col: "action_index",
+    output_count_col: "orchard_action_count",
+    note_reconstruction_cols: "rho, rseed",
+    shard_height: ORCHARD_SHARD_HEIGHT,
+};
+
+#[allow(dead_code)]
```

Comment:
AFAICT `ErrUnsupportedPool` is not unused. Why is the annotation necessary?

Reply:
The method it defines is unused when the `transparent-inputs` feature is not enabled. Rather than feature-flag the method of the trait, I thought it was better to just allow it to go unused.